src/identity/quartzOrganizations/reducers/index.ts

```diff
@@ -27,19 +35,35 @@ export default (state = initialState, action: Actions): QuartzOrganizations =>
       }
 
       case SET_QUARTZ_DEFAULT_ORG: {
-        const {oldDefaultOrgId, newDefaultOrgId} = action
-
-        if (oldDefaultOrgId !== newDefaultOrgId) {
-          draftState.orgs.forEach(org => {
-            if (org.id === oldDefaultOrgId) {
-              org.isDefault = false
-            }
-
-            if (org.id === newDefaultOrgId) {
-              org.isDefault = true
-            }
-          })
+        // No existing default org is acceptable; oldDefaultOrg may be undefined.
+        const oldDefaultOrg = draftState.orgs.find(
+          org => org.isDefault === true
+        )
+        const oldDefaultOrgId = oldDefaultOrg?.id
+        const {newDefaultOrgId} = action
+
+        if (oldDefaultOrgId === newDefaultOrgId) {
+          return
         }
+
+        let newIdCount = 0
+
+        draftState.orgs.forEach(org => {
+          if (org.id === oldDefaultOrgId) {
+            org.isDefault = false
+          }
+          if (org.id === newDefaultOrgId) {
+            org.isDefault = true
+            newIdCount++
+          }
+        })
+
+        if (newIdCount !== 1) {
+          throw new OrgValidationError(
+            'Error: Failed to identify new default org.'
+          )
+        }
```

Comment:
We almost certainly don't want to throw an error from a reducer, since they need to be pure, and throwing an error is a side effect (functional programming, amirite?) I think instead the reducer will need to set some kind of error state

Reply:
💯 I'll rewrite this so that it sets the status of this portion of the redux state to `RemoteDataState.Error`.